src/vulkan/device.cc

```diff
@@ -487,19 +487,29 @@ bool AreAllExtensionsSupported(
 }  // namespace
 
 Device::Device() = default;
-Device::Device(VkDevice device) : device_(device), destroy_device_(false) {}
+Device::Device(VkPhysicalDevice physical_device, VkDevice device)
+    : physical_device_(physical_device),
+      device_(device),
+      has_device_ownership_(false) {}
 Device::~Device() = default;
 
 void Device::Shutdown() {
-  if (destroy_device_) {
+  if (has_device_ownership_) {
     vkDestroyDevice(device_, nullptr);
     vkDestroyInstance(instance_, nullptr);
   }
 }
 
 Result Device::Initialize(const std::vector<Feature>& required_features,
                           const std::vector<std::string>& required_extensions) {
-  if (device_ == VK_NULL_HANDLE) {
+  if (has_device_ownership_ &&
+      (physical_device_ != VK_NULL_HANDLE || device_ != VK_NULL_HANDLE)) {
+    return Result(
+        "Vulkan: Device::Initialize Device class has VkDevice ownership but "
```

Comment:
I don't understand why this happens here, we don't pass physical_device_ or device_ into this method?